### PR TITLE
BUG: avoid circular import when pytest isn't available

### PR DESCRIPTION
**Pull request by kohr-h**

### CODE REVIEW

File: odl/tomo/util/testutils.py [hunk 1]
```diff
@@ -21,8 +21,9 @@
     import pytest
 
 except ImportError:
-    # Use the identity decorator (default of OptionalArgDecorator)
-    from odl.util import OptionalArgDecorator as identity
+
+    def identity(f, **kwargs):
```
Comment (adler-j) [inline odl/tomo/util/testutils.py]: And there is no other way to do this properly? This feels a bit like a hack.

Comment (kohr-h) [reply odl/tomo/util/testutils.py]: I'm not sure what you mean by "properly". We should have a function that can be used as a decorator and that does nothing. Actually, the `**kwargs` is wrong, there should be a little indirection.

Comment (kohr-h) [reply odl/tomo/util/testutils.py]: I changed the code to accommodate for optional args. Both of these will work and keep `f` untouched:
```python
@identity
def f(x):
    return x

@identity(a=42)
def f(x):
    return x
```

Comment (adler-j) [reply odl/tomo/util/testutils.py]: What I mean by properly is fixing the import :)

Comment (adler-j) [reply odl/tomo/util/testutils.py]: There is no way to do that? E.g. just reorder and put testutils last?

File: odl/util/testutils.py [hunk 2]
```diff
@@ -205,7 +205,12 @@ def is_subdict(subdict, dictionary):
     import pytest
 
 except ImportError:
-    from odl.util import OptionalArgDecorator as identity
+
+    def identity(*args, **kwargs):
```
Comment (adler-j) [inline odl/util/testutils.py]:
```
@identity(apply=lambda x: x+1)
def f(x):
    return 2 * x
```

This would fail, but perhaps that's expected?

Comment (kohr-h) [reply odl/util/testutils.py]: With the latest version it would "work" (do nothing).

File: odl/tomo/util/testutils.py (hunk 1 above)
Comment (kohr-h) [reply odl/tomo/util/testutils.py]: Maybe, but that import was overkill anyway. I'll leave it as it is currently.

**Comment by kohr-h** [comment]
Any objections to this version?

File: odl/util/testutils.py (hunk 2 above)
Comment (adler-j) [reply odl/util/testutils.py]: I guess the danger is 

```
@identity(lambda x: x+1)
def f(x):
    return 2 * x
```

But perhaps we just leave that be

**Comment by kohr-h** [comment]
> I guess the danger is
> 
>     @identity(lambda x: x+1)
>     def f(x):
>         return 2 * x
> 
> But perhaps we just leave that be

Yeah, it's really only intended as a drop-in so that the decorators don't crash anything if we cannot import `pytest`. The issue would be the same in the imported decorator anyway.

**Comment by adler-j** [comment]
Ok, I guess that's a hard corner case.

**Pull request merged**